### PR TITLE
Also publish a :debug docker image

### PR DESCRIPTION
**Pull request by eatonphil**
When you get a panic in a docker environment right now it's not easy to switch to a debug image to get a better stack trace.

So the idea here is to have a docker image like :latest but with the debug flag on. So `docker run ghcr.io/tigerbeetledb/tigerbeetle:debug`

### CODE REVIEW

File: docs/setup/with-docker.md [hunk 1]
```diff
@@ -25,3 +25,14 @@ info(main): 0: cluster=0: listening on 0.0.0.0:3000
 Now you can connect to the running server with any client. For a quick
 start, try [creating accounts and transfers in the Node
 CLI](../usage/node-cli).
+
+## Debugging panics
+
+If TigerBeetle panics and you can reproduce the panic, you can get a
+better stacktrace by switching to a debug image (by using the `:debug`
```
Comment (sentientwaffle) [inline docs/setup/with-docker.md]:
```suggestion
better stack trace by switching to a debug image (by using the `:debug`
```

File: docs/setup/with-docker-compose.md [hunk 2]
```diff
@@ -85,3 +82,13 @@ tigerbeetle_1    | info(clock): 1: system time is 78ns ahead
 Now you can connect to the running server with any client. For a quick
 start, try [creating accounts and transfers in the Node
 CLI](../usage/node-cli).
+
+## Debugging panics
+
+If TigerBeetle panics and you can reproduce the panic, you can get a
+better stacktrace by switching to a debug image (by using the `:debug`
```
Comment (sentientwaffle) [inline docs/setup/with-docker-compose.md]:
```suggestion
better stack trace by switching to a debug image (by using the `:debug`
```

File: scripts/install.sh [hunk 3]
```diff
@@ -1,7 +1,18 @@
 #!/bin/bash
+
+debug="$DEBUG"
+if [[ "$1" == "--debug" ]]; then
+    debug="true"
+fi
```
Comment (sentientwaffle) [inline scripts/install.sh:6]: Why have this flag in addition to the environment variable?

Comment (eatonphil) [reply scripts/install.sh:6]: The environment variable is necessary the way the docker build action works but flags are more natural on the cli IMO.

I can drop it though if you'd rather not have it.

Comment (sentientwaffle) [reply scripts/install.sh:6]: No, it is fine, I just wanted to make sure it wasn't left unintentionally.

File: docs/setup/with-docker.md (hunk 1 above)
Comment (eatonphil) [reply docs/setup/with-docker.md]: Fixed, thanks!

**Review by sentientwaffle — DISMISSED** [review]
LGTM. Maybe squash+rebase since there are a bunch of commits.

**Comment by eatonphil** [comment]
> LGTM. Maybe squash+rebase since there are a bunch of commits.

Is that different in effect than squash and merge?

**Comment by sentientwaffle** [comment]
Ah, that's what I meant :facepalm:

**Comment by eatonphil** [comment]
np! I try to always squash and merge

**Comment by eatonphil** [comment]
Oh sorry @sentientwaffle, I dismissed your review to make the `push` bit in line with the existing push. I don't actually understand what it means by `!= pull_request` since I think I saw other pull request branches in here. But I figured might as well keep it the same as the existing docker push condition.

**Review by sentientwaffle — APPROVED** [review]

**Comment by eatonphil** [comment]
Thanks!

**Pull request merged**